fs/fs_zenfs.cc

```diff
@@ -823,8 +859,40 @@ IOStatus ZenFS::RenameFileNoLock(const std::string& source_path,
       files_.insert(std::make_pair(source_path, source_file));
     }
   } else {
+    std::vector<std::string> children;
```

Comment:
This function is getting very long, can we move the new logic into a separate function  say, RenameDir() or something more apt, and call that function here.